dataSourceImpl/src/main/java/SharedPreferencesGatewayImpl.kt

```diff
@@ -0,0 +1,12 @@
+import model.PreferencesRepository
+
+class SharedPreferencesGatewayImpl(private val preferencesRepository: PreferencesRepository) :
```

Comment:
It's a wrong structure, Repository should depend on gateway, not vice versa. Dependencies order: PreferenceRepository -> SharedPreferenceGateway -> SharedPreferences;